navigator/runtime/src/main/java/com/freeletics/mad/navigator/internal/Navigate.kt

```diff
@@ -50,7 +66,7 @@ public fun navigate(
             (launcher as ActivityResultLauncher<Any?>).launch(event.permissions)
         }
         is NavEvent.DestinationResultEvent<*> -> {
-            controller.deliverResult(event.key, event.result)
+            savedStateHandleFor(event.key.route)[event.key.requestKey] = event.result
```

Comment:
What about still using `deliverResult` to not duplicate map access? or do you have other plans for it?
```suggestion
            deliverResult(event.key, event.result)
```

Reply:
I do plans to make the contents of key internal and move this and some other result based logic here, but that was meant for another pr